plugin.video.srgssr_ch_replay/resources/lib/srgssr_api_client/srgssr_api_client.py

```diff
@@ -0,0 +1,156 @@
+        """
+
+        if not creds["key"] or not creds["secret"]:
+            raise ValueError("SRGSSRApiClient creds must contain a 'key' and 'secret' fields")
+
+        self._base_url = base_url
+        self._basic_auth = HTTPBasicAuth(creds["key"], creds["secret"])
+        self._verify = verify
+        self._plugin = plugin
+        self._logger = self._plugin.logger
+        self._session = requests.Session()
+        self._auth_token = self.get_auth_token()
+
+    @property
+    def version(self):
+        if self._VERSION is None:
+            raise NotImplementedError(f"_VERSION attribute must be overriden in the subclass {self.__class__}")
+        return self._VERSION
+    
+    @property
+    def api_name(self):
+        if self._API_NAME is None:
+            raise NotImplementedError(f"_API_NAME attribute must be overriden in the subclass {self.__class__}")
+        return self._API_NAME
+
+    @property
+    def api_url_name(self):
+        if self._API_URL_NAME is None:
+            raise NotImplementedError(f"_API_URL_NAME attribute must be overriden in the subclass {self.__class__}")
+        return self._API_URL_NAME
+
+    def get_auth_token(self) -> str:
+        """Returning the authorization token
+        
+        Trying to get the auth token from the plugin settings. If it doesn't
+        exists, request a new one"""
+        token_exp = getattr(self._plugin.settings, f"srgssr_{self.api_name}_token_exp")
+        if token_exp:
+            token_exp = datetime.fromisoformat(token_exp)
+            token = getattr(self._plugin.settings, f"srgssr_{self.api_name}_token")
+            if token and (token_exp > datetime.utcnow()):
+                self._logger.debug(f"Using cached AuthToken (valid until {token_exp})")
+                return token
+
+        # Token doesn't exist or no more valid
+        return self.request_new_token()
+
+    def request_new_token(self) -> str:
+        self._logger.debug("Requesting new AuthToken")
+
+        params = {"grant_type": "client_credentials"}
+        res = requests.post(os.path.join(self._base_url, "oauth/v1/accesstoken"),params=params, auth=self._basic_auth)
+        if res.status_code in [401, 403]:
+            raise InvalidCredentialsException(self.api_name, "Invalid key/secret to access the API")
+        if not res.ok:
+            raise SRGSSRApiException(self.api_name, f"Error{res.status_code}: {res.content}")
+        
+        data = res.json()
+        token = data["access_token"]
+        expires_in = data["expires_in"]
+        token_exp = (datetime.utcnow() + timedelta(seconds=expires_in)).isoformat()
+        self._logger.debug(f"Got a new AuthToken valid until {token_exp}")
+
+        setattr(self._plugin.settings, f"srgssr_{self.api_name}_token", token)
+        setattr(self._plugin.settings, f"srgssr_{self.api_name}_token_exp", token_exp)
+
+        return token
+
+    def _renew_access_token(func):
```

Comment:
got 4 PyLint issues here (rest of the code is good)
![image](https://user-images.githubusercontent.com/20797220/164803550-c0b2b876-d4d1-496f-afb6-88a196f815ef.png)